File: together-backend/src/main/java/org/mafisher/togetherbackend/service/FriendService.java
package org.mafisher.togetherbackend.service;

import org.mafisher.togetherbackend.dto.UserDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.security.Principal;
import java.util.List;

public interface FriendService {
    void sendRequest(String nickname, Principal principal);
    void acceptRequest(String nickname, Principal principal);
    void rejectRequest(String nickname, Principal principal);
    List<UserDto> getFriends(Principal principal, Pageable pageable);
    List<UserDto> getReceivedRequests(Principal principal, Pageable pageable);
    Page<UserDto> searchUsers(String query, Pageable pageable, Principal principal);
}

File: together-frontend/src/api/friendService.js
import apiClient from './apiClient';

export const sendRequest = async (nickname) => {
    const response = await apiClient.post(`/friends/requests/${nickname}`);
    return response.data;
};

export const acceptRequest = async (nickname) => {
    const response = await apiClient.post(`/friends/requests/${nickname}/accept`);
    return response.data;
};

export const rejectRequest = async (nickname) => {
    const response = await apiClient.post(`/friends/requests/${nickname}/reject`);
    return response.data;
};

export const getFriends = async (pageable) => {
    const response = await apiClient.get('/friends/list', {
        params: {
            page: pageable.page,
            size: pageable.size,
            sort: pageable.sort
        }
    });
    return response.data;
};

export const getReceivedRequests = async (pageable) => {
    const response = await apiClient.get('/friends/requests/received', {
        params: {
            page: pageable.page,
            size: pageable.size,
            sort: pageable.sort
        }
    });
    return response.data;
};

export const searchUsers = async (query, pageable) => {
    const response = await apiClient.get('/friends/search', {
        params: {
            query: query,
            page: pageable.page,
            size: pageable.size,
            sort: pageable.sort
        }
    });
    return response.data;
};